src/client/common/application/commands.ts

```diff
@@ -149,4 +148,7 @@ export interface ICommandNameArgumentTypeMapping extends ICommandNameWithoutArgu
     [DSCommands.SubmitGitHubIssue]: [];
     [DSCommands.ShowDataViewer]: [IShowDataViewerFromVariablePanel];
     [DSCommands.ClearSavedJupyterUris]: [];
+    [DSCommands.SelectJupyterURI]: [undefined, 'toolbar' | undefined];
+    [DSCommands.SelectNativeJupyterUriFromStatusBar]: [];
+    [DSCommands.SelectNativeJupyterUriFromToolBar]: [];
```

Comment:
We seem to have a set of commands with arguments and a set without. Seems like these two new ones don't have arguments. But...does this matter? Not sure if I see a reason for the distinction.

Reply:
Also, why | undefined? Isn't 'commandPalette' the other option?

Reply:
This is because `SelectJupyterUri` command can only be invoked from the command palette in which case the argument is `undefiend`.
The only other place its used is Interactive & WebView notebooks where we pass the argument `toolbar` to indicate its been invoked from the interactive/webview notebook toolbar.

Other than that there are no places where this command is invoked from.

We have similar commands that do the exact same thing, launch the UI. However we have separate commands so we can uniquely identify how and where those commands are launched. Pity we cannot use the same command with arguments (in VSC toolbar and statusbars).
I know in treeviews we can use commands with args.

Reply:
Hope that makes sense. The complication comes from the fact that we want to identify the differnt places the same command is launched, hence duplicate commands.

Reply:
Oh wait, just found we can use args in the statusbar.

Reply:
Fixed by removing one of the duplicate commands.